### PR TITLE
feat: Create film extension

### PR DESCRIPTION
**Pull request by amfage**
Add the `film` STAC Extension, made up of a explanatory README, JSON Schema, examples and non-examples.

### CODE REVIEW

File: extensions/film/README.md
Comment (dwsilk) [inline extensions/film/README.md:16]: Removed reference to NZAM. How about

> Identifier assigned to each roll of film.

File: extensions/film/schema.json
Comment (l0b0) [inline extensions/film/schema.json:31]: This shouldn't be necessary, as done in https://github.com/linz/stac/pull/58. Basically, if we're validating against this schema then the file contains a reference to this extension, so it's tautological to validate that we're referencing the extension.

File: extensions/film/README.md
Comment (dwsilk) [inline extensions/film/README.md:17]: Maybe we should call this `negative_id` ...

**Review by dwsilk — CHANGES_REQUESTED** [review]

File: extensions/film/schema.json
Comment (dwsilk) [reply extensions/film/schema.json:31]: Nice, I thought that looked weird.

File: extensions/film/examples/item.json
Comment (l0b0) [inline extensions/film/examples/item.json:25]: Would you be interested in creating any examples of [non-conformant files](https://github.com/linz/stac/pull/56/files)? They are a good way of verifying that the schema actually does what it says.

File: extensions/film/schema.json
Comment (blacha) [reply extensions/film/schema.json:31]: I disagree with this, I also believe #58 should not have been merged. We need to have wider review of sweeping changes before merging things..

By not having the schema referenced in the extensions  you could validate a random stac file against this schema and it would be considered "valid" even though it is invalid as stac_extension was not referenced.

All of the STAC spec extensions I looked at have this reference, I do not think we should be deviating from their standard.

eg https://github.com/stac-extensions/raster/blob/main/json-schema/schema.json#L72

Comment (l0b0) [reply extensions/film/schema.json:31]: Why would you validate a file against a schema which is not referenced within that file?

Comment (l0b0) [reply extensions/film/schema.json:31]: Asked [upstream](https://github.com/stac-extensions/template/issues/10).

Comment (blacha) [reply extensions/film/schema.json:31]: Because that's how most JSON schemas work? JSON documents generally never reference the schema they are being validated against.

I would also try and keep as close to the STAC examples as possible.

Comment (l0b0) [reply extensions/film/schema.json:31]: > Because that's how most JSON schemas work? JSON documents generally never reference the schema they are being validated against.

STAC has an explicit mechanism for referencing the schemas though. I would've thought not referencing the schema would mean the file isn't a STAC document.

Comment (dwsilk) [reply extensions/film/schema.json:31]: I don't really want this issue to hold up this PR. Sounds like https://github.com/linz/stac/pull/58 should be reversed pending reasoning/advice from STAC Spec maintainers, with discussion reopened on a new PR to remove if necessary?

File: extensions/film/README.md
Comment (dwsilk) [reply extensions/film/README.md:17]: Let's call it `negative_sequence`.

File: extensions/film/examples/item.json
Comment (dwsilk) [reply extensions/film/examples/item.json:25]: Yep we can do that. A `no_film_id.json`, a `no_negative_sequence.json`, maybe a `negative_sequence_string.json` ..

File: extensions/film/schema.json
Comment (l0b0) [reply extensions/film/schema.json:31]: Why not leave it as is? Nothing's broken, and it's not possible to write a negative test for that part of the old schema (which is how this got started).

Comment (dwsilk) [reply extensions/film/schema.json:31]: I think the default thing we should do when concern is raised over a merged PR (especially one that is yet to be released) is to revert and discuss.

Comment (dwsilk) [reply extensions/film/schema.json:31]: The [current example](https://github.com/linz/stac/blob/41a13241e28e3521cf072a947398eabacc590415/extensions/linz/examples/collection.json) of a valid `collection.json` on the `linz` extension doesn't reference the `quality` extension in `stac_extensions`. Should it? Would that have been picked up if the `schema.json` for the `linz` extension contained this check?

File: extensions/film/examples/item.json
Comment (amfage) [reply extensions/film/examples/item.json:25]: I've added 4 non-examples, missing field and incorrect type for both required fields

**Review by dwsilk — APPROVED** [review]

File: extensions/film/schema.json
Comment (l0b0) [reply extensions/film/schema.json:37]: It doesn't because it doesn't need to, and therefore shouldn't, as explained [upstream](https://github.com/stac-extensions/template/issues/10). Having to list nested extensions is pointless overhead.

Comment (dwsilk) [reply extensions/film/schema.json:37]: Why does it list the `version` extension?

Comment (l0b0) [reply extensions/film/schema.json:37]: Because, like the template repo, the version extension requires us to reference it in every STAC file even though it's referenced from the LINZ extension.

**Pull request merged**
